app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java

```diff
@@ -215,7 +247,8 @@ private void addNewDevice(final Intent intent) {
         final BluetoothDevice device = intent.getParcelableExtra(BluetoothDevice.EXTRA_DEVICE);
 
         // Skip paired devices and devices without a name.
-        if (device.getBondState() != BluetoothDevice.BOND_BONDED && !device.getName().equals("null")) {
+        if (device != null && device.getName() != null
+            && device.getBondState() != BluetoothDevice.BOND_BONDED && !device.getName().equals("null")) {
```

Comment:
@TheBonheurs I'm not sure, but I think that PMD error does not complain about private methods (at least I don't think it should)